### DISCUSSION TITLE
How to *prevent* installation on ARM64

### DISCUSSION THREAD

**Discussion by dorssel** [microsoft/winget-pkgs#176095] [Q&A] [answered]
My package only has a x64 installer: https://github.com/microsoft/winget-pkgs/blob/master/manifests/d/dorssel/usbipd-win/4.3.0/dorssel.usbipd-win.installer.yaml

It turns out that this can be installed on ARM64 as well, as it has x64 emulation. However, the package contains drivers, which only work on x64. ARM64 users install the package, only to find that it doesn't work on their architecture after all. Is there a way to prevent installation on any architecture other than the supported x64?

And as a side question, for those familiar with MSI. Is there a way to detect the *real* architecture from within the MSI installer, such that it can block installation?

**Comment by mdanish-kh** [comment, marked as answer]
You can add `UnsupportedOSArchitectures` to the installer manifest and it _should_ block users on the listed architectures from installing the app.

```diff
PackageIdentifier: dorssel.usbipd-win
PackageVersion: 4.3.0
InstallerLocale: en-US
InstallerType: wix
Scope: machine
+ UnsupportedOSArchitectures:
+  - arm64
+  - arm
InstallerSwitches:
  Custom: /norestart
[...]
```
